src/enumerable.rb

```diff
@@ -314,6 +314,20 @@ def find(if_none = nil)
     nil
   end
 
+  def filter_map
+    return enum_for(:filter_map) unless block_given?
+
+    gather = ->(item) { item.size <= 1 ? item.first : item }
+
+    arr = []
+    each do |*item|
+      if yield(gather.(item))
+        arr << yield(gather.(item))
```

Comment:
I could be wrong, but I don’t think we should yield to the block twice. What if the block has side effects?

Reply:
Yup, no reason to call yield twice here.